### PR TITLE
Documentation, features, and unit tests for Yukawa

### PR DESCRIPTION
**Pull request by adisr04**
- Add Yukawa documentation
- Add _derivative method, fix other calculations
- Add all unit tests for Yukawa in test_pair

### CODE REVIEW

File: relentless/potential/pair.py [hunk 1]
```diff
@@ -156,32 +156,151 @@ def _derivative(self, param, r, epsilon, sigma, **params):
             d = d.item()
         return d
 
-""" TODO: these potentials need updating
 class Yukawa(PairPotential):
+    """Yukawa pair potential.
+
+    .. math::
+
+    u(r) = \varepsilon\frac{{e}^{{-\kappa{r}}}}{{r}}
```
Comment (mphoward) [inline relentless/potential/pair.py:164]: This needs another indent and can be written like:
```suggestion
    u(r) = \varepsilon\frac{e^{-\kappa r}}{r}
```

File: relentless/potential/pair.py [hunk 2]
```diff
@@ -156,32 +156,151 @@ def _derivative(self, param, r, epsilon, sigma, **params):
...
+
+    The required coefficients per pair are:
+
+    - :math:`\varepsilon`: interaction energy scale
```
Comment (mphoward) [inline relentless/potential/pair.py:168]:
```suggestion
    - :math:`\varepsilon`: prefactor (dimensions: energy x length)
```

File: relentless/potential/pair.py [hunk 3]
```diff
@@ -156,32 +156,151 @@ def _derivative(self, param, r, epsilon, sigma, **params):
...
+    - :math:`\kappa`: screening length inverse
```
Comment (mphoward) [inline relentless/potential/pair.py:169]:
```suggestion
    - :math:`\kappa`: inverse screening length
```

File: relentless/potential/pair.py [hunk 4]
```diff
@@ -156,32 +156,151 @@ def _derivative(self, param, r, epsilon, sigma, **params):
             d = d.item()
         return d
 
-""" TODO: these potentials need updating
 class Yukawa(PairPotential):
```
Comment (zeesherman) [inline relentless/potential/pair.py:159]: Add `Yukawa` to `__all__` in line 1?

**Pull request merged**